pkgs/development/interpreters/python/python-packages-base.nix

```diff
@@ -37,15 +37,21 @@ let
   else
     ./python2/mk-python-derivation.nix;
 
-  buildPythonPackage = makeOverridablePythonPackage (lib.makeOverridable (callPackage mkPythonDerivation {
+  makeFinalAttrsCompatible = fun: attrs:
```

Comment:
I don't know the details, but I have heard that implementation of the finalAttrs pattern is being considered at the lower level.
@emilazy may know about this.
https://matrix.to/#/!VjfUzaKsXokUdnQcvP:nixos.org/$VEPTUoA8AX14mLeSmf62MVwgnuMlLUagdeHOpWi8ZhU

Also, this kind of change should be an independent PR.

Reply:
See:

* https://github.com/NixOS/nixpkgs/pull/234651
* https://github.com/NixOS/nixpkgs/pull/271387

Reply:
(Personally I’d rather we move away from custom builder functions to hooks and avoid the need to reimplement all the functionality in every one. But that’s a bigger change.)

Reply:
Removed in latest force-push.